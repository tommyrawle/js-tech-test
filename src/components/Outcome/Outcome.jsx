import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Price from '../Price/Price.container.jsx';
import { getDisplayableOutcomes } from '../../redux/selectors';
import { OutcomeWrapper } from './Outcome.styles.jsx';

const Outcome = ({ outcomeId, outcomes }) => {
  const outcome = outcomes.find(outcome => outcome.outcomeId === outcomeId);

  if (outcome) {
    return (
      <OutcomeWrapper>
        {outcome.name}
        {<Price price={outcome.price} />}
      </OutcomeWrapper>
    );
  } else return null;
};

Outcome.propTypes = {
  outcomeId: PropTypes.number,
  outcomes: PropTypes.arrayOf(PropTypes.object)
};

const mapStateToProps = state => ({
  outcomes: getDisplayableOutcomes(state)
});

export default connect(mapStateToProps)(Outcome);
