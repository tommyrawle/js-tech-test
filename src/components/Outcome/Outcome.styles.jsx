import styled from 'styled-components';

export const OutcomeWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 5px;
  border-bottom: 1px solid #ececec;
`;
