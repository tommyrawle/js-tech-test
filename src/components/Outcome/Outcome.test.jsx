import React from 'react';
import { shallow } from 'enzyme';
import { Outcome } from './Outcome.jsx';

describe('Outcome component', () => {
  it('should render Outcome component', () => {
    const mockProps = {
      outcomeId: 789,
      outcomes: [{ outcomeId: 789, name: 'something or other', price: { den: 7, num: 1, decimal: 1.1428571428571428 } }]
    };
    expect(shallow(<Outcome {...mockProps} />)).toMatchSnapshot();
  });
  it('should render null if outcome does not exist', () => {
    const mockProps = {
      outcomeId: 789,
      outcomes: []
    };
    expect(shallow(<Outcome {...mockProps} />)).toMatchSnapshot();
  });
});
