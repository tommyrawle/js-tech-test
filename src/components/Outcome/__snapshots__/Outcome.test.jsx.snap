// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Outcome component should render Outcome component 1`] = `
<styled.div>
  something or other
  <Connect(Price)
    price={
      Object {
        "decimal": 1.1428571428571428,
        "den": 7,
        "num": 1,
      }
    }
  />
</styled.div>
`;

exports[`Outcome component should render null if outcome does not exist 1`] = `""`;
